src/test/scala/uk/gov/nationalarchives/omega/jms/JmsRRSqsClientSpec.scala

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2023 The National Archives
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package uk.gov.nationalarchives.omega.jms
+
+import cats.effect.std.Queue
+import cats.effect.testing.scalatest.AsyncIOSpec
+import cats.effect.{IO, Resource}
+import org.scalatest.FutureOutcome
+import org.scalatest.freespec.FixtureAsyncFreeSpec
+import org.scalatest.matchers.must.Matchers
+import org.typelevel.log4cats.SelfAwareStructuredLogger
+import org.typelevel.log4cats.slf4j.Slf4jFactory
+import uk.gov.nationalarchives.omega.jms.JmsRRClient.ReplyMessageHandler
+
+class JmsRRSqsClientSpec extends FixtureAsyncFreeSpec with AsyncIOSpec with Matchers {
+
+  override type FixtureParam = JmsRRClient[IO]
+
+  override def withFixture(test: OneArgAsyncTest): FutureOutcome = {
+    // setup for cats-effect
+    val logging = Slf4jFactory[IO]
+    implicit val logger: SelfAwareStructuredLogger[IO] = logging.getLogger
+    val replyQueue = "omega-editorial-web-application-instance-1"
+    // Create a Jms Request-Reply Client (e.g. from the Play Application start up method)
+    val clientRes: Resource[IO, JmsRRClient[IO]] = JmsRRClient.createForSqs[IO](
+      HostBrokerEndpoint("localhost", 9324),
+      UsernamePasswordCredentials("x", "x"),
```

Comment:
These creds don't seem to matter, true?

Reply:
True, these are the default creds for ElasticMQ - which are public knowledge.

Reply:
Oddly enough, it doesn't matter what you put it there.

Reply:
Oh OK. Well I guess it is something needed for the SQS API but which ElasticMQ doesn't care about.